### ISSUE TITLE
Redis cluster installation error when using redis-operator 0.10.2

### ISSUE THREAD

**Issue by danil-smirnov** [OT-CONTAINER-KIT/helm-charts#37]
Hi,

I've installed the latest `redis-operator` chart:

```
$ helm ls -aA

NAME                   	NAMESPACE        	REVISION	UPDATED                               	STATUS  	CHART                          	APP VERSION
redis-operator         	redis-operator   	1       	2022-04-27 13:14:30.963308 +0300 EEST 	deployed	redis-operator-0.10.2          	0.10.0
```

But the cluster installation command is failing then:

```
$ helm install test ot-helm/redis-cluster --set redisCluster.clusterSize=3 --namespace redis-operator

Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(RedisCluster.spec.kubernetesConfig): missing required field "serviceType" in in.opstreelabs.redis.redis.v1beta1.RedisCluster.spec.kubernetesConfig
```
Maybe you can introduce some tests to ensure the new version is usable?

**Comment by bill-ding007** [comment]
I use redis-operator 0.10.2 have same error information:
```
ERROR   controller-runtime.manager.controller.rediscluster  Reconciler error        {"reconciler group": "redis.redis.opstreelabs.in", "reconciler kind": "RedisCluster", "name": "redis-cluster", "namespace": "redis", "error": "Service \"redis-cluster-leader-headless\" is invalid: [spec.ipFamily: Invalid value: \"null\": field is immutable, spec.ipFamily: Required value]"}
```
Has it been officially tested?

**Comment by bill-ding007** [comment]
> Hi,
> 
> I've installed the latest `redis-operator` chart:
> 
> ```
> $ helm ls -aA
> 
> NAME                   	NAMESPACE        	REVISION	UPDATED                               	STATUS  	CHART                          	APP VERSION
> redis-operator         	redis-operator   	1       	2022-04-27 13:14:30.963308 +0300 EEST 	deployed	redis-operator-0.10.2          	0.10.0
> ```
> 
> But the cluster installation command is failing then:
> 
> ```
> $ helm install test ot-helm/redis-cluster --set redisCluster.clusterSize=3 --namespace redis-operator
> 
> Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(RedisCluster.spec.kubernetesConfig): missing required field "serviceType" in in.opstreelabs.redis.redis.v1beta1.RedisCluster.spec.kubernetesConfig
> ```
> 
> Maybe you can introduce some tests to ensure the new version is usable?



> Hi,
> 
> I've installed the latest `redis-operator` chart:
> 
> ```
> $ helm ls -aA
> 
> NAME                   	NAMESPACE        	REVISION	UPDATED                               	STATUS  	CHART                          	APP VERSION
> redis-operator         	redis-operator   	1       	2022-04-27 13:14:30.963308 +0300 EEST 	deployed	redis-operator-0.10.2          	0.10.0
> ```
> 
> But the cluster installation command is failing then:
> 
> ```
> $ helm install test ot-helm/redis-cluster --set redisCluster.clusterSize=3 --namespace redis-operator
> 
> Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(RedisCluster.spec.kubernetesConfig): missing required field "serviceType" in in.opstreelabs.redis.redis.v1beta1.RedisCluster.spec.kubernetesConfig
> ```
> 
> Maybe you can introduce some tests to ensure the new version is usable?

Is it solved? I'm having the same problem.

**Comment by iamabhishek-dubey** [comment]
This is fixed in 0.10.3

**Comment by ljuaneda** [comment]
![image](https://user-images.githubusercontent.com/10739749/174844950-a2bc61bf-0948-427e-a0e7-9407dfd3f3d3.png)
same issue with 0.10.3
